src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,149 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+import duke.data.Storage;
+import duke.data.TaskList;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.user.DukeException;
+import duke.user.Parser;
+import duke.user.Ui;
+
+/**
+ * Main class for the Duke program.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+
+    /**
+     * Constructor for the Duke instance.
+     *
+     * @param filePath the String representing the path of the file where the data is saved
+     */
+    public Duke(String filePath) {
+        storage = new Storage(filePath);
+        tasks = new TaskList(storage.loadFromFile());
+        ui = new Ui();
+    }
+
+    /**
+     * Default constructor for Duke.
+     * Required for JavaFX GUI
+     */
+    public Duke() {
+        storage = new Storage("data/Duke.txt");
+        tasks = new TaskList(storage.loadFromFile());
+        ui = new Ui();
+    }
+
+    /**
+     * Handles the user input and sends it to the parser to perform the commands, if valid.
+     */
+    public String getResponse(String input) {
+
+        Parser parser = new Parser(tasks);
+        boolean taskListIsUpdated = false;
+        boolean taskListIsAddedTo = false;
+        String output = "";
+
+        try {
+            String[] parsedInputString = parser.checkInput(input);
+            String command = parsedInputString[0];
+
+            // Case where user marks a task as done
+            if (command.equals("done")) {
+                Task current = tasks.getTask(Integer.parseInt(parsedInputString[1]) - 1);
+                current.setIsDone();
+                output += ui.displayDoneTaskMessage(current.toString());
+                taskListIsUpdated = true;
+
+                // case where user wants to delete a task item, similar to done
```

Comment:
Good job on writing comments to make your code easier to read. Should you capitalize `case` so that it is consistent with the other comments?